Boop/Boop/scripts/lib/turndown.js

```diff
@@ -0,0 +1,924 @@
+var TurndownService = (function () {
```

Comment:
Ok, with `module.exports = (function () {` it loads, but failing on document:
```
[HTML To Markdown] Error: ReferenceError: Can't find variable: document 
```

I tried to find, but it seems all HTML to Markdown libraries are using `document`/`window` to parse HTML for it it to then output as Markdown (at least for what I checked in this [list](https://stackoverflow.com/a/40066280/1030110)).

Reply:
Yeah, I was thinking it might be possible to use jsdom to parse the HTML, but that would require quite an extensive rewrite. Prehaps somthing like [mmd.js](https://github.com/p01/mmd.js) is a more realistic option. It doesn't support some markdown features but its tiny and is just a bunch of string manipulations so it should be easy to adapt for Boop.

Reply:
Hey there! @mrbenshef is correct, you'll need a library that can run outside of a browser... Anything advertised as working in Node.js should technically work (unless it spawns a headless browser or something...).

If you can find one that can go back and forth between MD and HTML, thatway we won't need to add multiple libraries.

Reply:
The only bi-directional library I found is [`showdown`](https://github.com/showdownjs/showdown). But it requires JSDom or something similar.

Reply:
Fair enough. To be honest I'm kind of surprised to see HTML to Markdown was requested, I'd assume the opposite would be much more common (and easier to implement)...

I found this, which uses regexes to perform the substitution. If it can't be sued both ways, then include it in the script itself rather than as a library: https://github.com/thetutlage/html-to-markdown

Reply:
@IvanMathy I wanted to tackle both to/from after I saw the #141 request.
And "Markdown To HTML" is much easier: #155